backend/server.ts

```diff
@@ -1,38 +1,65 @@
 import express from 'express';
-import {User, UserName} from "../src/modules/users/user-types";
+import {Skill, User, UserName, UserSkill} from "../src/modules/users/user-types";
 
 const port = 3001;
 const app = express();
 
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
-let users: User[] = [{id: 0, firstName: 'Donald', lastName: 'Trump'}];
+const skills: Skill[] = [{id: 'skill-0', name: 'Politics'},{id: 'skill-1', name: 'Sword Fighting'}, {id: 'skill-2', name: 'Dragon Riding'}];
+const users: User[] = [{id: 'user-0', firstName: 'Donald', lastName: 'Trump', regnalNumber: 1, skills: [{skill: skills[0], level: 9999}]}];
 
 app.get('/users', (req, res) => {
    res.json(users);
 });
 
+const isUserName = (userName: unknown): userName is UserName => (
+    typeof userName === 'object' &&
+       userName !== null &&
+       'firstName' in userName &&
+       'lastName' in userName
+);
+
 app.post('/users', (req, res) => {
-   const firstName = req.body.firstName;
-   const lastName = req.body.lastName;
+   const userName = req.body;
 
-   if (firstName === undefined || lastName === undefined) {
+   if (!isUserName(userName)) {
       res.sendStatus(400);
+      return;
    }
-   else {
-      const newUser: User = {
-         id: users.length,
-         firstName: firstName,
-         lastName: lastName
-      };
 
-      users = [...users, newUser];
+   const firstName = userName.firstName;
+   const lastName = userName.lastName;
```

Comment:
Tohle jde zjednodušit na
```suggestion
   const {firstName, lastName} = userName;
```